websocket/src/websocket.cpp

```diff
@@ -386,6 +419,82 @@ static int LuaSend(lua_State* L)
     return 0;
 }
 
+void HttpServerRequestCallback(void* user_data, wsHttpServer::Request* request)
+{
+    if (g_Websocket.m_Connections.Size() >= g_Websocket.m_Server->m_Connections.Capacity()) {
+        request->m_CloseConnection = 1;
+        return;
+    }
+    request->m_RemoveConnection = 1;
+
+    WebsocketConnection* conn = ConvertToConnection(request->m_Socket);
+    conn->m_ConnectTimeout = dmTime::GetTime() + 3000 * 1000;
+    conn->m_CustomHeaders = 0;
+    conn->m_Protocol = 0;
+
+    conn->m_Callback = g_Websocket.m_ServerCallback;
+
+    if (g_Websocket.m_Connections.Full())
+        g_Websocket.m_Connections.OffsetCapacity(2);
+    g_Websocket.m_Connections.Push(conn);
+
+    SetState(conn, STATE_SERVER_HANDSHAKE_READ);
+}
+
+static int LuaListen(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+
+    if (!g_Websocket.m_Initialized)
+        return DM_LUA_ERROR("The web socket module isn't initialized");
+
+    if (g_Websocket.m_Server)
+        return DM_LUA_ERROR("The web server is already initialized");
+
+    const int port = luaL_checkinteger(L, 1);
+    const int max_connections = luaL_checkinteger(L, 2);
+    const int connection_timeout = luaL_checkinteger(L, 3);
+    g_Websocket.m_ServerCallback = dmScript::CreateCallback(L, 4);
```

Comment:
Add error if  m_ServerCallback  is 0.

You don't need server without cb. For example If you create server when require lua model. You don't have script instance and cb will be 0. (i spend 2h trying to understand why my server not get messages)